language-plutus-core/src/Language/PlutusCore/Name.hs

```diff
@@ -74,6 +76,17 @@ newIdentifier str = do
             put (IM.insert key str is, M.insert str nextU ss, nextU')
             pure nextU
 
+-- | Add to the 'ByteString' representation of a 'Name' its 'Unique'
+-- without any additional symbols inbetween.
```

Comment:
The way we do it when debug printing is with an underscore before the unique - can we be consistent?

Reply:
Actually, I'm not sure I understand why we're doing this at all - why can't we just leave the unique where it is?

Reply:
This is used only in limited circumstances to get output like

`(all s0 (size) (fun [(con integer) s0] (fun [(con integer) s0] [(con integer) s0])))`

I don't think

`(all s_0 (size) (fun [(con integer) s_0] (fun [(con integer) s_0] [(con integer) s_0])))`

would be nicer. Also the whole point of the function is to make the unique an actual part of the name rather than emphasis those are still distinct things via `_`. Also if anyone uses debug printing over `s0` he'll get `s0_0` which is ugly, but tolerable, and with `s_0` he'll get `s_0_0` which is really ugly.

So I don't think this should be changed.

Reply:
> Also the whole point of the function is to make the unique an actual part of the name

I don't understand why you're doing that. We have overlapping names in a bunch of other places, and yeah, they look weird unless you use debug printing to disambiguate them. That's just the approach we're taking atm, let's at least be consistent instead of sometimes trying to get around it.

Reply:
i.e. what's wrong with just dropping the whole `revealUnique` thing?

Reply:
You get

`(all s (size) (all s (size) (fun [(con size) s] (fun [(con integer) s] [(con integer) s]))))`

instead of

`(all s0 (size) (all s1 (size) (fun [(con size) s1] (fun [(con integer) s0] [(con integer) s1]))))`

I highly prefer the latter.

Reply:
So you need to use debug printing - just like any of the rest of our types and terms! I agree that it's annoying to have lots of name shadowing in terms and then need to use debug printing to disambiguate, but I'd prefer that we had that problem either consistently or not, rather than sometimes working around it.

Reply:
My current rule of thumb is that if I generate names recursively from the same string representation, then I add uniques. Otherwise not.

This type can be a part of a large program and we'll be forced to print every single unique in that program. Moreover

`(all s0 (size) (all s1 (size) (fun [(con size) s1] (fun [(con integer) s0] [(con integer) s1]))))`

is what I'd read by hand and what is written in spec, but... In a large program that will be a big index and it'll look silly. Yeah, you're right, I'll fix that in a dumb, but principled way.

Reply:
I've just been relying on the debug printing. It's not that bad IMO!

Reply:
Okay, now it generates what I would write by hand.